### PR TITLE
added support for projection selection

### LINKED ISSUE #43: Support multiple projections in the map
Right now the map is hard coded to work with WGS84 (EPSG:4326). We would like this option to be configurable to support other projections like Web mercator (EPSG:3857).

### PR DESCRIPTION
**Pull request by CptSchnitz**
closes #43

### CODE REVIEW

**Comment by alebinson** [comment]
Do you run it to see mouse tracking work as expected?

**Comment by CptSchnitz** [comment]
Mouse tracking is working and shows WGS84 coordinates.

File: packages/react-components/src/lib/ol-map/map.tsx [hunk 1]
```diff
@@ -79,9 +84,9 @@ export const Map: React.FC<MapProps> = (props) => {
             format(
               coord as Coordinate,
               '{y}°N {x}°E',
-              COORDINATES_FRACTION_DIFITS
+              COORDINATES_FRACTION_DIGITS
             ),
-          projection: PROJECTION,
+          projection: Proj.WSG84,
```
Comment (alebinson) [inline packages/react-components/src/lib/ol-map/map.tsx]: Probably should depends on MAP projection, advice with @syncush

Comment (CptSchnitz) [reply packages/react-components/src/lib/ol-map/map.tsx]: If we want to support multiple projections here we need more work on the format. also it shouldn't be based on the map as usually you want WGS84.

Comment (syncush) [reply packages/react-components/src/lib/ol-map/map.tsx]: We should have a format for each projection, the `MousePosition` and `Map` projection should be independent.
It is the programmer's responsibility to match the implementation to the needs.

Comment (alebinson) [reply packages/react-components/src/lib/ol-map/map.tsx]: @syncush Actually I'm looking for unification. "Consumer" gives me a PROJECTION_TYPE and we correspondingly use proper mouse tracking presentation. Tracking ability is internals, from my point of view

Comment (alebinson) [reply packages/react-components/src/lib/ol-map/map.tsx]: @syncush @CptSchnitz think we should transfer **center** and  **zoom** as well . Current params should "play" as DEFAULT ones

Comment (syncush) [reply packages/react-components/src/lib/ol-map/map.tsx]: The component is made for us and for 3rd party developers (A.K.A Consumers) and as developers, we should be flexible as much as possible but our application should be strict. 

@CptSchnitz  what do you think?

Comment (CptSchnitz) [reply packages/react-components/src/lib/ol-map/map.tsx]: I do think we need to provide the option to set the center an zoom from outside, to be as flexible as possible, but i do think in that case we should separate the view into its own component.
If we do want it we should open a new issue as its out of scope of this one.

Comment (syncush) [reply packages/react-components/src/lib/ol-map/map.tsx]: please create proper issues and link this PR and Issue to them.

**Review by syncush — APPROVED** [review]

File: packages/react-components/src/lib/ol-map/map.tsx (hunk 1 above)
Comment (alebinson) [reply packages/react-components/src/lib/ol-map/map.tsx]: as I already wrote to @CptSchnitz: "mostly at this time I want to make things easy and unified for us(not on 3rd party)".
As much as we open our components for outside configuration as much we'll see duplicated code in APPs.
Let's think about common needs and incapsulate them inside our components (less freedom)

Comment (syncush) [reply packages/react-components/src/lib/ol-map/map.tsx]: @alebinson for example `VectorTilesLayer` doesn't support translating from WGS84 to Mercator and the other way around, so we have to pass it as an argument.

File: packages/react-components/src/lib/ol-map/map.tsx [hunk 2]
```diff
@@ -75,20 +102,15 @@ export const Map: React.FC<MapProps> = (props) => {
...
       map.addControl(
         new MousePosition({
-          coordinateFormat: (coord): string =>
-            format(
-              coord as Coordinate,
-              '{y}°N {x}°E',
-              COORDINATES_FRACTION_DIFITS
-            ),
-          projection: PROJECTION,
+          coordinateFormat: getCoordinateFormatString(projection),
```
Comment (alebinson) [inline packages/react-components/src/lib/ol-map/map.tsx]: think default value should be **projection ?? Proj.WGS84**

**Review by alebinson — CHANGES_REQUESTED** [review]

File: packages/react-components/src/lib/ol-map/map.tsx (hunk 2 above)
Comment (CptSchnitz) [reply packages/react-components/src/lib/ol-map/map.tsx]: fixed

**Review by alebinson — APPROVED** [review]

**Pull request merged**